Add section on dotnet watch behavior with response compression (#32481)
dotnet/docs: docs/core/tools/dotnet-watch.md

While running `dotnet watch`, you can force the app to rebuild and restart by pressing Ctrl+R in the command shell. This feature is available only while the app is running. For example, if you run `dotnet watch` on a console app that ends before you press Ctrl+R, pressing Ctrl+R has no effect. However, in that case `dotnet watch` is still watching files and will restart the app if a file is updated.

### Response compression

> If `dotnet watch` runs for an app that uses [response compression](/aspnet/core/performance/response-compression), the tool can't inject the browser refresh script. In that case, the .NET 7 version of the tool displays a warning message like the following:

Reviewer: ```suggestion
If `dotnet watch` runs for an app that uses [response compression](/aspnet/core/performance/response-compression), the tool can't inject the browser refresh script. The  .NET 7 and later version of the tool displays a warning message like the following:
```

Reviewer: Now you don't need a moniker.

> warn: Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware[4]
>
> Unable to configure browser refresh script injection on the response. This may have been caused by the response's Content-Encoding: 'br'. Consider disabling response compression.

> As an alternative to disabling response compression, you can manually add the browser refresh JavaScript reference to the app's pages:

Reviewer: ```suggestion
As an alternative to disabling response compression, manually add the browser refresh JavaScript reference to the app's pages:
```
per style guide, shorter is better.
